pandas/core/indexes/base.py

```diff
@@ -800,24 +800,29 @@ def __deepcopy__(self, memo=None):
             memo = {}
         return self.copy(deep=True)
 
-    def _validate_names(self, name=None, names=None, deep=False):
+    def _handle_names(self, name=None, names=None, deep=False):
```

Comment:
I see the original issue, but how does this fix things, ``_validate_names`` is stilling doing partial validation and partial conversions.

Reply:
Well, I agree with that to some extent. But conversion can be removed if Pandas deprecates `name` argument for `MultiIndex` as it was proposed in https://github.com/pandas-dev/pandas/pull/19168#issuecomment-357176138 .

Reply:
well, no, the reason for acceptance is that MI needs to have the same interface as Index.

Reply:
But if `Index` accepts `names` as an argument, they will be compatible.